logstash-core/lib/logstash/pipeline.rb

```diff
@@ -147,7 +147,7 @@ module LogStash; class Pipeline < BasePipeline
 
   MAX_INFLIGHT_WARN_THRESHOLD = 10_000
 
-  def initialize(config_str, settings = SETTINGS, namespaced_metric = nil)
+  def initialize(config_str, settings = SETTINGS, namespaced_metric = nil, agent = nil)
```

Comment:
I don't think this work with the current master because the agent is never sent to the pipeline?  See https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/pipeline_action/create.rb#L40